macros/math/nondecimal_base.pl

```diff
@@ -0,0 +1,147 @@
+################################################################################
+# WeBWorK Online Homework Delivery System
+# Copyright &copy; 2000-2022 The WeBWorK Project, https://github.com/openwebwork
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of either: (a) the GNU General Public License as published by the
+# Free Software Foundation; either version 2, or (at your option) any later
+# version, or (b) the "Artistic License" which comes with this package.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See either the GNU General Public License or the
+# Artistic License for more details.
+################################################################################
+
+=head1 NAME
+
+nondecimal_base.pl - Handles conversions to and from non-decimal bases.
+
+=head1 DESCRIPTION
+
+The subroutine C<convertBase> converts nubmers to and from bases up to hexadecimal.
+
+    $x = 47;
+    $x2 = convertBase($x, to => 2);
+
+    $y16 = convertBase('2EF9', from => 16);
+
+This can be used in problems with base conversion.
+
+    $x = random(200,500);
+    $x_16 = Real(convertBase($x, to => 16));
```

Comment:
You should make it clear that this requires the `ArbitraryStrings` context.

Reply:
Yes.  very true.